core/src/main/clojure/xtdb/pgwire.clj

```diff
@@ -994,6 +992,34 @@
     (cmd-write-msg conn msg-backend-key-data {:process-id (:cid conn), :secret-key 0})
     (cmd-send-ready conn)))
 
+(defn user+password [node user]
+  (first (xt/q node "SELECT passwd AS password, username AS user FROM pg_user WHERE username = ?" {:args [user]})))
+
+(defn cmd-startup-pg30 [conn startup-params]
+  (let [{:keys [node conn-state !closing?]} conn]
+    (if node
```

Comment:
There is sort of a chicken/egg problem for the playground as I need a node to get the password, but the node only gets attached after startup. I could do some rejuggling if we want authn for the playground as well or always skip authn for the playground. Might be a bit weird to not be able to test authn via the playground.